packages/cashier/src/pages/p2p-cashier/p2p-cashier.tsx

```diff
@@ -109,9 +114,41 @@ const P2PCashier = ({
                 setOrderId(input_order_id);
             }
         },
-        [history, location.hash, location.search, order_id]
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [history, location.hash, location.search]
```

Comment:
@hamid-deriv Any reason for omitting `order_id` from the dependency array?

Reply:
I remove this Because it causes [this](https://redmine.deriv.cloud/issues/76458#The_redirecting_url_when_we_place_an_order_should_have_order_number.In_the_test_link_it_is_showing_wrong) issue.
Before (in js), the callback was called after `useEffect`, so changing `order_id` will not affect the route query, after migrating to TS, the setQueryOrder should be defined before `useEffect`, and having order_id as a dependency will remove the query.